internal/core/src/index/InvertedIndexTantivy.cpp

```diff
@@ -447,6 +331,176 @@ InvertedIndexTantivy<T>::BuildWithRawData(size_t n,
     }
 }
 
+template <typename T>
+void
+InvertedIndexTantivy<T>::build_index(
+    const std::vector<std::shared_ptr<FieldDataBase>>& field_datas) {
+    switch (schema_.data_type()) {
+        case proto::schema::DataType::Bool: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<bool>(static_cast<const bool*>(data->Data()),
+                                         n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Int8: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<int8_t>(
+                    static_cast<const int8_t*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Int16: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<int16_t>(
+                    static_cast<const int16_t*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Int32: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<int32_t>(
+                    static_cast<const int32_t*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Int64: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<int64_t>(
+                    static_cast<const int64_t*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Float: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<float>(
+                    static_cast<const float*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Double: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<double>(
+                    static_cast<const double*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::VarChar: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<std::string>(
+                    static_cast<const std::string*>(data->Data()), n);
+            }
+            break;
+        }
```

Comment:
why not directly using T replace type
 for (const auto& data : field_datas) {
                auto n = data->get_num_rows();
                wrapper_->add_data<T>(
                    static_cast<const T*>(data->Data()), n);
            }
            break;

Reply:
OK.